azurerm/internal/services/kusto/kusto_table_resource.go

```diff
@@ -0,0 +1,339 @@
+package kusto
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/Azure/azure-kusto-go/kusto"
+	dataplaneKusto "github.com/Azure/azure-kusto-go/kusto"
+	"github.com/Azure/azure-kusto-go/kusto/data/table"
+	"github.com/Azure/azure-kusto-go/kusto/data/types"
+	"github.com/Azure/azure-kusto-go/kusto/unsafe"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/kusto/parse"
+	dataplaneTypes "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/kusto/types"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/kusto/validate"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+)
+
+func resourceArmKustoTable() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmKustoTableCreateUpdate,
+		Read:   resourceArmKustoTableRead,
+		Update: resourceArmKustoTableCreateUpdate,
+		Delete: resourceArmKustoTableDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
```

Comment:
we should check the id format just like other resources do

Reply:
Could you be more precise what you would change here?

Reply:
for example: https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/azurerm/internal/services/appplatform/spring_cloud_app_resource.go#L28-L31

Reply:
Fixed